src/ChorusMerge/Program.cs

```diff
@@ -40,24 +40,6 @@ public static int Main(string[] args)
 				string commonFilePath = args[1];
 				string theirFilePath = args[2];
 
-				if (Platform.IsWindows)
-				{
-					if (!RuntimeInformation.FrameworkDescription.Contains("Framework"))
-					{
-						Encoding.RegisterProvider(CodePagesEncodingProvider.Instance); // required for .NET6
-					}
-
-					// Convert the input arguments from cp1252 -> utf8 -> ucs2
-					// It always seems to be 1252, even when the input code page is actually something else. CP 2012-03
-					// var inputEncoding = Console.InputEncoding;
-					var inputEncoding = Encoding.GetEncoding(1252);
```

Comment:
I tested that rather thoroughly, and what happens is we get the correctly-encoded Unicode filenames in `string[] args`. I changed RepositorySetup.ProjectNameForTest to ไก่ (the Thai word for chicken, also used in the Main_Utf8FilePaths_FileNamesOk test that you commented on below. I then added code into Program.cs to print out the input character-by-character, with codepoints. And what we got was the three characters U+0E44 THAI CHARACTER SARA AI MAIMALAI (that's the ไ), U+0E01 THAI CHARACTER KO KAI (the ก), and U+0E48 THAI CHARACTER MAI EK (a combining accent mark, the vertical line above the ก). All correct and exactly as they should be.

Note that this is also *after* removing a whole swathe of code from fixutf8.py that was using the Win32 process API to launch external processes like the merge tool (ChorusMerge). That may have been necessary in Python 2, but Python 3.6 changed things. Now, the standard Python library's `subprocess.call` and `subprocess.Popen`, which everyone (including Mercurial) uses to launch external processes, will pass Unicode arguments correctly. So Mercurial tells Python "here are some Unicode strings" (filenames) "to put on the command line". Python encodes it correctly, the C# runtime receives that and correctly interprets it, and Program.cs receives an array of correctly-encoded strings (which are UTF-16 at that point, though I haven't confirmed it by running non-BMP emojis through the system yet).

So yes, we thankfully no longer need to "wrap" UTF-8 characters in CP-1252 encoding to make them work. And in fact, wrapping UTF-8 characters in CP-1252 is not actually 100% safe! It's mostly safe, but the very string I happened to use for testing is one of the failure modes. See, U+0E01 THAI CHARACTER KO KAI (ก, the first letter of the Thai alphabet) encodes in UTF-8 as 0xE0 0xB8 0x81. If you then try to decode that as CP-1252, you'll discover that 0x81 is actually **invalid** in codepage 1252! [There's no character at 0x81 in CP-1252](https://en.wikipedia.org/wiki/Windows-1252#Codepage_layout). Some decoding methods will pass 0x81 through to U+0081, but others will replace it with the undefined character or throw an exception like UnicodeDecodeError. I came across that one in my testing, which is what led me to find out that the UTF-8-wrapped-in-CP-1252 was actually completely unnecessary: I commented it out once just to see what would happen to the error, and to my shock everything worked fine! (Except for the Main_Utf8FilePaths_FileNamesOk test, which you commented on below, which was actually testing the wrapping that I had just gotten rid of).

Reply:
I changed RepositorySetup.ProjectNameForTest to be "a📚aไก่a projéct" instead of "test projéct", so that it would include spaces, accented Roman characters, non-Roman characters, *and* astral-plane characters (📚 is U+1F4DA, which will be encoded with a surrogate pair in UTF-16). I then ran the tests, and they continued to pass. (Test run is still ongoing as I'm writing this, but all the tests that fail when I disable the fixutf8 extension, thereby proving that they are affected by encoding issues, have passed, so I feel confident that the rest of the tests will pass as well).

So I'm quite confident that non-Roman characters are getting passed to ChorusMerge correctly: I've tested every scenario that usually breaks things, and they all worked.